### PR TITLE
CI/TST: Making ci/run_tests.sh fail if one of the steps fail

### PR DESCRIPTION
**Pull request by datapythonista**
Looks like when simplifying the running of the tests in the CI (#23924), I missed the `-e` in the bash header. And that makes the `ci/run_tests.sh` exit with status code 0, even if the calls to pytests fail.

This left the CI in green, even when tests fail for the last 3 days (sorry about that). I think nothing is broken.

This PR fixes the problem.

CC: @pandas-dev/pandas-core

### CODE REVIEW

**Comment by gfyoung** [comment]
Seems reasonable to me.  Wondering though if we should double check by adding a failing test just to test that this fails as expected?

**Comment by datapythonista** [comment]
I tested locally. I'd personally merge this asap, as the test of all PRs are failing silently (unless you check the logs). And we'll see quickly if this works as expected as soon as other PR tests fail.

But I surely can break something here and test if you think it's worth the delay.

**Comment by gfyoung** [comment]
> But I surely can break something here and test if you think it's worth the delay.

Fair enough.  I think we all can be vigilant to notice if it doesn't work. 🙂

**Review by gfyoung — APPROVED** [review]
Let's merge on green.

**Comment by TomAugspurger** [comment]
Any idea what the failure is from @datapythonista? https://travis-ci.org/pandas-dev/pandas/jobs/462954286#L1845

```
$ ci/run_tests.sh
============================= test session starts ==============================
platform linux2 -- Python 2.7.15, pytest-4.0.1, py-1.7.0, pluggy-0.8.0
rootdir: /home/travis/build/pandas-dev/pandas, inifile: setup.cfg
plugins: xdist-1.24.1, forked-0.2, hypothesis-3.68.0
gw0 [0]
scheduling tests via LoadScheduling
 generated xml file: /home/travis/build/pandas-dev/pandas/test-data-single.xml -
======================== no tests ran in 84.95 seconds =========================
The command "ci/run_tests.sh" exited with 5.
```

Both of those failures are the slow / single tests, right?

**Comment by TomAugspurger** [comment]
I see. Do we ever want to run `single and slow` though? I would have thought that it'd be `single or slow`.

**Comment by TomAugspurger** [comment]
So to run all the SLOW jobs, we need

`-n 1 -m slow and single`

and then

`-n 2 -m slow and not single`


And the problem is that `-m slow and single` is empty? As a workaround, can we add a dummy test with both those marks?

**Comment by datapythonista** [comment]
That's correct @TomAugspurger, looks like it's a common pytest issue. The pytest devs propose the workaround I pushed. Not the nicest code, but this should work for now, and I'll try to find a cleaner approach for that script later (I don't like calling pytest twice, will try to avoid that).

**Comment by TomAugspurger** [comment]
LGTM.

**Comment by datapythonista** [comment]
Can't really understand the errors in travis. They are surely unrelated, they don't seem to happen in master, and I can't reproduce them locally. @TomAugspurger have you seen this error before?
```
_____________________ TestTSPlot.test_is_error_nozeroindex _____________________
[gw0] linux -- Python 3.6.6 /home/travis/miniconda3/envs/pandas-dev/bin/python
self = <pandas.tests.plotting.test_datetimelike.TestTSPlot object at 0x7fca6c1875c0>
    def test_is_error_nozeroindex(self):
        # GH11858
        i = np.array([1, 2, 3])
        a = DataFrame(i, index=i)
>       _check_plot_works(a.plot, xerr=a)
pandas/tests/plotting/test_datetimelike.py:89: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/tests/plotting/test_datetimelike.py:1562: in _check_plot_works
    pickle.dump(fig, path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
m = <bound method AffineBase.transform of <matplotlib.transforms.BlendedAffine2D object at 0x7fca6c567240>>
    def _pickle_method(m):
        """Handle pickling issues with class instance."""
>       if m.im_self is None:
E       AttributeError: 'function' object has no attribute 'im_self'
../../../miniconda3/envs/pandas-dev/lib/python3.6/site-packages/statsmodels/graphics/functional.py:32: AttributeError
```

**Comment by datapythonista** [comment]
That's weird, not sure why matplotlib or statmodels should fail randomly.

In any case, all green now. I'll let someone else review it before merging, as I made some changes since last reviews.

File: ci/run_tests.sh
Comment (jreback) [inline ci/run_tests.sh]: if no tests are found it is an error

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/run_tests.sh
Comment (datapythonista) [reply ci/run_tests.sh]: we don't have tests for the builds when we filter by "slow", when we execute the "single" call here. So, all our slow builds will fail.

Comment (jreback) [reply ci/run_tests.sh]: ok thanks
even more explicit comments here would help then

but can be more another pass

Comment (jreback) [reply ci/run_tests.sh]: sure

**Pull request merged**
